integration/sawtooth_integration/tests/test_permission.py

```diff
@@ -0,0 +1,254 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import logging
+import unittest
+import traceback
+import tempfile
+import os
+import subprocess
+import time
+from sawtooth_cli.main import main
+import sys
+from io import StringIO
+
+from sawtooth_integration.tests.integration_tools import wait_for_rest_apis
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.StreamHandler())
+LOGGER.setLevel(logging.DEBUG)
+
+TEST_WIF = '5Jq6nhPbVjgi9vTUuK7e2W81VT5dpQR7qPweYJZPVJKNzSornyv'
+TEST_PUBKEY = \
+    '033775c26a68a3872f03314ccd080b8d8ec828572469737c7d3aa467f853a069d5'
+
+
+class TestPermission(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        wait_for_rest_apis(['rest_api:8080'])
+
+    def setUp(self):
+        self._temp_dir = tempfile.mkdtemp()
+        self._data_dir = os.path.join(self._temp_dir, 'data')
+        os.makedirs(self._data_dir)
+
+        # create a wif key for signing
+        self._wif_file = os.path.join(self._temp_dir, 'test.priv')
+        with open(self._wif_file, 'wb') as wif:
+            wif.write(TEST_WIF.encode())
+
+    def _run(self, args):
+        try:
+            LOGGER.debug("Running %s", " ".join(args))
+            proc = subprocess.run(args,
+                                  stdout=subprocess.PIPE,
+                                  stderr=subprocess.PIPE,
+                                  check=True)
+            LOGGER.debug(proc.stdout.decode())
+        except subprocess.CalledProcessError as err:
+            LOGGER.debug(err)
+
+    def _read_from_stdout(self, cmd, args):
+        # Retrieve string of setting contents for comparison to input settings
+        backup = sys.stdout  # backup the environment
+        sys.stdout = StringIO()  # Capture the output of next statement
+
+        main(cmd, args)
+        settings = sys.stdout.getvalue()  # release the output and store
+        sys.stdout.close()
+        # Restore the environment
+        sys.stdout = backup
+        return settings
+
+    def _try_to_get_output(self, cmd, args, required):
+        # Temporary function until --wait is added to sawtooth config
+        for i in range(10):
+            settings = self._read_from_stdout(cmd, args)
+            if required in settings:
+                return settings
+            time.sleep(1)
+        self.fail("Changes were not committed in 5 seconds.")
+
+    def _assert_line_starts_match(self, raw_output, expected_line_starts):
+        output_lines = raw_output.split('\n')
+
+        for output, expected in zip(output_lines, expected_line_starts):
+            self.assertTrue(
+                output.startswith(expected),
+                'List output "{}" did not match "{}"'.format(expected, output))
```

Comment:
I think this is my fault, but `output` and `expected` should be switched in the format args.